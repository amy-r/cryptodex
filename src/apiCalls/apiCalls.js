import apiKey from './apiKey';

export const fetchApi = async () => {
  try {
    const response = await fetch(`https://api.nomics.com/v1/dashboard?key=${apiKey}`)
    const currData = await response.json()
    console.log(currData)
  } catch (error) {
    throw new Error("error getting data")
  }
}
