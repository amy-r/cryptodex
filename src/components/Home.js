import React, { Component } from 'react';
import { connect } from 'react-redux';

class Home extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      amount: null
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    
  }

  handleChange = (event) => {
    const { value, name } = event.target;
    this.setState({ [name]:value })
  }

  render() {
    return(
      <div>
        <form onSubmit= {this.handleSubmit}>
          <input type='text' name='name' placeholder='name' value={this.state.name} onChange={this.handleChange}/>
          <input type='text' name='amount' placeholder='amount' value={this.state.amount} onChange={this.handleChange}/>
          <input type='submit' />
        </form>  
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  dashboard: state.dashboard
}


export default connect(mapStateToProps, null)(Home);