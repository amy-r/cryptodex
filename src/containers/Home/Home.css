h1 {
  color: white;
  position: relative;
  z-index: 10;
  display: inline;
}

.logout {
  float: right;
  position: relative;
  z-index: 11;
  display: inline;
}

form {
  position: relative;
  z-index: 12
}