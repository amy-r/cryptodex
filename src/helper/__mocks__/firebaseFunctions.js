export const writeCurrency = jest.fn();

export const getUserData = jest.fn().mockImplementation( () => {
  Promise.resolve( {BTC:1, ETH:1, then: jest.fn()})
})