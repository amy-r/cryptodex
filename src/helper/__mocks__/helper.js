/* eslint-disable */

import {mockDashboard} from '../../helper/mockDashboard';

export const createMarketCap = jest.fn().mockImplementation( () => {
  return [
      { 'currency' : "BTC",
        'close' : "7436.82313",
        'availableSupply' : "16957550.0",
        'marketCap' : 126110300068.1315 }
    ]
})

export const sortByCaps = jest.fn().mockImplementation( () => {
  return mockDashboard
})

export const trimData = jest.fn().mockImplementation( () => {
  return mockDashboard
})


export const createCapPercentage = jest.fn().mockImplementation( () => {
  return mockDashboard
})

export const removeMismatches = jest.fn().mockImplementation( () => {
  return { BTC: 1, ETH: 1}
})

export const findMismatches = jest.fn().mockImplementation( () => {
  return { FAKECOIN: 1 }
})

export const calculateValue = jest.fn().mockImplementation( () => {
  return [7436.82, 387.43,]
})

export const calculatePercent = jest.fn().mockImplementation( () => {
  return [6.67,13.33,20,26.67,33.33]
})