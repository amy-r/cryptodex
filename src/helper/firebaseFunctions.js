import * as firebase from 'firebase';
import { config } from '../firebase/firebase';
export const firebaseApp = firebase.initializeApp(config);

export const writeCurrency = (user, curr) => {
  try {
    firebaseApp.database().ref(user).update({
      [curr.name]: curr.amount,
    });
  } catch (error) {
    throw new Error(`error adding currency : ${error}` )
  }
}

