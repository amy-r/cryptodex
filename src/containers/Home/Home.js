import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as firebase from 'firebase';
import { addCurrency } from '../../actions/addCurrency';
import { addUser } from '../../actions/addUser';
import { removeUser } from '../../actions/removeUser';
import { getUserPortfolio } from '../../actions/getUserPortfolio';
import { withRouter } from 'react-router-dom';
import { writeCurrency, getUserData } from '../../helper/firebaseFunctions.js';
import UserPortfolio from '../UserPortfolio/UserPortfolio';
import TopTen from '../TopTen/TopTen';
import './Home.css';
import PropTypes from 'prop-types';

export class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: 'BTC',
      amount: ''
    }
  }

  componentDidMount = () => {
    firebase.auth().onAuthStateChanged( (user) => {
      if (user) {
        const displayName = user.displayName;
        const uid = user.uid;
        const userInfo = { displayName, uid};
        this.props.logIn(userInfo);     
        getUserData(uid).then((portfolio) => this.props.getUserPortfolio(portfolio));
      } else {
        this.props.history.push('/');
      }
    }) 
  }

  handleSubmit = (event) => {
    const { addCurrency, user } = this.props;
    event.preventDefault();
    const newCurrency = {...this.state}
    addCurrency(newCurrency);
    writeCurrency(user.uid, newCurrency);
    this.setState({name: '', amount: ''});
    getUserData(user.uid).then((portfolio) => this.props.getUserPortfolio(portfolio));
  }

  handleChange = (event) => {
    const { value, name } = event.target;
    this.setState({ [name]:value });
  }

  logOut = () => {
    firebase.auth().signOut();
    this.props.logOut();
    this.props.history.push('/');
  }

  createOptions = () => {
    const { dashboard } = this.props
    const options = dashboard.map( (curr) => {
      return <option key={curr.currency} value={curr.currency}> {curr.currency} </option>
    })
    return options
  }

  render() {
    return (
      <div>
        <header>
          <h1> <span className='crypto'>CRYPTO</span><span className='dex'>DEX</span></h1>
          <button className='logout' onClick= {this.logOut}> Log Out </button>     
        </header>
        <div className='body'>
          <div className= 'left'>
            <UserPortfolio />
            <h4> Add a currency you own </h4>
            <form onSubmit= {this.handleSubmit}>
              <select name='name' onChange={this.handleChange}>
               {this.createOptions()}
              </select>  
              <input type='text' name='amount' placeholder='amount' value={this.state.amount} onChange={this.handleChange}/>
              <input type='submit' />
            </form>
          </div>  
          <div className='right'>
            <h2> Today's Market </h2>  
            <TopTen />
          </div>
        </div>    
      </div>
    )   
  }
}

export const mapStateToProps = (state) => ({
  dashboard: state.dashboard,
  user: state.user,
  portfolio: state.portfolio
});

export const mapDispatchToProps = (dispatch) => ({
  addCurrency: (currency) => dispatch(addCurrency(currency)),
  logIn: (user) => dispatch(addUser(user)),
  logOut: () => dispatch(removeUser()),
  getUserPortfolio: (portfolio) => dispatch(getUserPortfolio(portfolio))
});

Home.propTypes = {
  logIn: PropTypes.func,
  addCurrency: PropTypes.func,
  getUserPortfolio: PropTypes.func,
  portfolio: PropTypes.objectOf(PropTypes.string),
  dashboard: PropTypes.arrayOf(PropTypes.objectOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number]))),
  user: PropTypes.shape({
    displayName: PropTypes.string,
    uid: PropTypes.string,
  })
};


export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Home));