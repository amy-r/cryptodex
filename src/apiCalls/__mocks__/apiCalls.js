/* eslint-disable */

import { mockDashboard } from '../../helper/mockDashboard';

export const fetchApi = jest.fn().mockImplementation( () => {
  return Promise.resolve(mockDashboard);
});