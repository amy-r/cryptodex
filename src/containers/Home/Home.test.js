/* eslint-disable */

import React from 'react';
import { Home, mapDispatchToProps, mapStateToProps } from './Home.js';
import { shallow } from 'enzyme';
import { fetchApi } from '../../apiCalls/apiCalls.js';
import * as helper from '../../helper/helper';
import {mockDashboard} from '../../helper/mockDashboard'; 
import { writeCurrency, getUserData } from '../../helper/firebaseFunctions.js';

jest.mock('../../helper/firebaseFunctions')
describe('Home', () => {
  let mockUser;
  let mockPortfolio;
  let wrapper;

  beforeEach( () => {
    mockUser = 1;
    mockPortfolio = {BTC: 1, ETH: 1}
    wrapper = shallow(<Home dashboard={mockDashboard} user={mockUser} portfolio={mockPortfolio} history={[]} getUserPortfolio={jest.fn()} addCurrency={jest.fn()}  />, {disableLifecycleMethods: true})
  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('calls login on comonentDidMount', () => {
    expect(wrapper.props().logIn).toHaveBeenCalled;
  })

  it('calls getUserData on comonentDidMount', () => {
    expect(getUserData).toHaveBeenCalled;
  })

  it('calls getUserPortfolio on comonentDidMount', () => {
    expect(wrapper.props().getUserPortfolio).toHaveBeenCalled;
  })

  it.skip('calls addCurrency on handleSubmit', async () => {
    const mockEvent = {preventDefault: jest.fn()}
    wrapper.instance().handleSubmit(mockEvent)
    expect(wrapper.props().addCurrency).toHaveBeenCalled 
  })

  it('sets state on handleChange', () => {
    const mockEvent = { target: { name: 'name', value: 'ETH' }}
    expect(wrapper.state().name).toEqual('BTC')
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state().name).toEqual('ETH')
  })
})

describe('MSTP and MDTP', () => {
  it('should define dashboard props for the container', () => {
    const mockStore = { dashboard: mockDashboard }
    const mapped = mapStateToProps(mockStore)
    expect(mapped.dashboard).toEqual(mockDashboard)
  })

  it('should define user props for the container', () => {
    const mockUser = 1
    const mockStore = {user: mockUser}
    const mapped = mapStateToProps(mockStore)
    expect(mapped.user).toEqual(mockUser)
  })

  it('should define portfolio props for the container', () => {
    const mockPortfolio = {ETH:1, BTC:1}
    const mockStore = {portfolio: mockPortfolio}
    const mapped = mapStateToProps(mockStore)
    expect(mapped.portfolio).toEqual(mockPortfolio)
  })

  it('should call dispatch on addCurrency', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addCurrency();
    expect(mockDispatch).toHaveBeenCalled;
  })

  it('should call dispatch on logIn', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.logIn();
    expect(mockDispatch).toHaveBeenCalled;
  }) 

  it('should call dispatch on getUserPortfolio', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.getUserPortfolio();
    expect(mockDispatch).toHaveBeenCalled;
  })   
})