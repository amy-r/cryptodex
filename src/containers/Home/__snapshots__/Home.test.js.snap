// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    addCurrency={[Function]}
    dashboard={
        Array [
            Object {
              "availableSupply": "16957550.0",
              "currency": "BTC",
              "marketCap": 126110300068.1315,
              "price": "7436.82313",
            },
            Object {
              "availableSupply": "98680455.0",
              "currency": "ETH",
              "marketCap": 38207824855.0488,
              "price": "387.43253",
            },
            Object {
              "availableSupply": "39094520623.0",
              "currency": "XRP",
              "marketCap": 19161006447.74476,
              "price": "0.49032",
            },
          ]
    }
    getUserPortfolio={[Function]}
    history={Array []}
    portfolio={
        Object {
            "BTC": "1",
            "ETH": "1",
          }
    }
    user={
        Object {
            "displayName": "Oprah",
            "uid": "1",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
                     
                    <span
                              className="crypto"
                    >
                              CRYPTO
                    </span>
                    <span
                              className="dex"
                    >
                              DEX
                    </span>
          </h1>
          <button
                    className="logout"
                    onClick={[Function]}
          >
                     Log Out 
          </button>
</header>,
        <div
          className="body"
>
          <div
                    className="left"
          >
                    <Connect(UserPortfolio) />
                    <h4>
                               Add a currency you own 
                    </h4>
                    <form
                              onSubmit={[Function]}
                    >
                              <select
                                        name="name"
                                        onChange={[Function]}
                              >
                                        <option
                                                  value="BTC"
                                        >
                                                   
                                                  BTC
                                                   
                                        </option>
                                        <option
                                                  value="ETH"
                                        >
                                                   
                                                  ETH
                                                   
                                        </option>
                                        <option
                                                  value="XRP"
                                        >
                                                   
                                                  XRP
                                                   
                                        </option>
                              </select>
                              <input
                                        name="amount"
                                        onChange={[Function]}
                                        placeholder="amount"
                                        type="text"
                                        value=""
                              />
                              <input
                                        type="submit"
                              />
                    </form>
          </div>
          <div
                    className="right"
          >
                    <h2>
                               Today's Market 
                    </h2>
                    <Connect(Component) />
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
               
              <span
                            className="crypto"
              >
                            CRYPTO
              </span>
              <span
                            className="dex"
              >
                            DEX
              </span>
</h1>,
            <button
              className="logout"
              onClick={[Function]}
>
               Log Out 
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                <span
                  className="crypto"
>
                  CRYPTO
</span>,
                <span
                  className="dex"
>
                  DEX
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CRYPTO",
                  "className": "crypto",
                },
                "ref": null,
                "rendered": "CRYPTO",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "DEX",
                  "className": "dex",
                },
                "ref": null,
                "rendered": "DEX",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Log Out ",
              "className": "logout",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": " Log Out ",
            "type": "button",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="left"
>
              <Connect(UserPortfolio) />
              <h4>
                             Add a currency you own 
              </h4>
              <form
                            onSubmit={[Function]}
              >
                            <select
                                          name="name"
                                          onChange={[Function]}
                            >
                                          <option
                                                        value="BTC"
                                          >
                                                         
                                                        BTC
                                                         
                                          </option>
                                          <option
                                                        value="ETH"
                                          >
                                                         
                                                        ETH
                                                         
                                          </option>
                                          <option
                                                        value="XRP"
                                          >
                                                         
                                                        XRP
                                                         
                                          </option>
                            </select>
                            <input
                                          name="amount"
                                          onChange={[Function]}
                                          placeholder="amount"
                                          type="text"
                                          value=""
                            />
                            <input
                                          type="submit"
                            />
              </form>
</div>,
            <div
              className="right"
>
              <h2>
                             Today's Market 
              </h2>
              <Connect(Component) />
</div>,
          ],
          "className": "body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(UserPortfolio) />,
                <h4>
                   Add a currency you own 
</h4>,
                <form
                  onSubmit={[Function]}
>
                  <select
                                    name="name"
                                    onChange={[Function]}
                  >
                                    <option
                                                      value="BTC"
                                    >
                                                       
                                                      BTC
                                                       
                                    </option>
                                    <option
                                                      value="ETH"
                                    >
                                                       
                                                      ETH
                                                       
                                    </option>
                                    <option
                                                      value="XRP"
                                    >
                                                       
                                                      XRP
                                                       
                                    </option>
                  </select>
                  <input
                                    name="amount"
                                    onChange={[Function]}
                                    placeholder="amount"
                                    type="text"
                                    value=""
                  />
                  <input
                                    type="submit"
                  />
</form>,
              ],
              "className": "left",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Add a currency you own ",
                },
                "ref": null,
                "rendered": " Add a currency you own ",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <select
                      name="name"
                      onChange={[Function]}
>
                      <option
                                            value="BTC"
                      >
                                             
                                            BTC
                                             
                      </option>
                      <option
                                            value="ETH"
                      >
                                             
                                            ETH
                                             
                      </option>
                      <option
                                            value="XRP"
                      >
                                             
                                            XRP
                                             
                      </option>
</select>,
                    <input
                      name="amount"
                      onChange={[Function]}
                      placeholder="amount"
                      type="text"
                      value=""
/>,
                    <input
                      type="submit"
/>,
                  ],
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <option
                          value="BTC"
>
                           
                          BTC
                           
</option>,
                        <option
                          value="ETH"
>
                           
                          ETH
                           
</option>,
                        <option
                          value="XRP"
>
                           
                          XRP
                           
</option>,
                      ],
                      "name": "name",
                      "onChange": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": "BTC",
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            " ",
                            "BTC",
                            " ",
                          ],
                          "value": "BTC",
                        },
                        "ref": null,
                        "rendered": Array [
                          " ",
                          "BTC",
                          " ",
                        ],
                        "type": "option",
                      },
                      Object {
                        "instance": null,
                        "key": "ETH",
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            " ",
                            "ETH",
                            " ",
                          ],
                          "value": "ETH",
                        },
                        "ref": null,
                        "rendered": Array [
                          " ",
                          "ETH",
                          " ",
                        ],
                        "type": "option",
                      },
                      Object {
                        "instance": null,
                        "key": "XRP",
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            " ",
                            "XRP",
                            " ",
                          ],
                          "value": "XRP",
                        },
                        "ref": null,
                        "rendered": Array [
                          " ",
                          "XRP",
                          " ",
                        ],
                        "type": "option",
                      },
                    ],
                    "type": "select",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "name": "amount",
                      "onChange": [Function],
                      "placeholder": "amount",
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                   Today's Market 
</h2>,
                <Connect(Component) />,
              ],
              "className": "right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Today's Market ",
                },
                "ref": null,
                "rendered": " Today's Market ",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
                         
                        <span
                                    className="crypto"
                        >
                                    CRYPTO
                        </span>
                        <span
                                    className="dex"
                        >
                                    DEX
                        </span>
            </h1>
            <button
                        className="logout"
                        onClick={[Function]}
            >
                         Log Out 
            </button>
</header>,
          <div
            className="body"
>
            <div
                        className="left"
            >
                        <Connect(UserPortfolio) />
                        <h4>
                                     Add a currency you own 
                        </h4>
                        <form
                                    onSubmit={[Function]}
                        >
                                    <select
                                                name="name"
                                                onChange={[Function]}
                                    >
                                                <option
                                                            value="BTC"
                                                >
                                                             
                                                            BTC
                                                             
                                                </option>
                                                <option
                                                            value="ETH"
                                                >
                                                             
                                                            ETH
                                                             
                                                </option>
                                                <option
                                                            value="XRP"
                                                >
                                                             
                                                            XRP
                                                             
                                                </option>
                                    </select>
                                    <input
                                                name="amount"
                                                onChange={[Function]}
                                                placeholder="amount"
                                                type="text"
                                                value=""
                                    />
                                    <input
                                                type="submit"
                                    />
                        </form>
            </div>
            <div
                        className="right"
            >
                        <h2>
                                     Today's Market 
                        </h2>
                        <Connect(Component) />
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                 
                <span
                                className="crypto"
                >
                                CRYPTO
                </span>
                <span
                                className="dex"
                >
                                DEX
                </span>
</h1>,
              <button
                className="logout"
                onClick={[Function]}
>
                 Log Out 
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <span
                    className="crypto"
>
                    CRYPTO
</span>,
                  <span
                    className="dex"
>
                    DEX
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "CRYPTO",
                    "className": "crypto",
                  },
                  "ref": null,
                  "rendered": "CRYPTO",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "DEX",
                    "className": "dex",
                  },
                  "ref": null,
                  "rendered": "DEX",
                  "type": "span",
                },
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Log Out ",
                "className": "logout",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": " Log Out ",
              "type": "button",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="left"
>
                <Connect(UserPortfolio) />
                <h4>
                                 Add a currency you own 
                </h4>
                <form
                                onSubmit={[Function]}
                >
                                <select
                                                name="name"
                                                onChange={[Function]}
                                >
                                                <option
                                                                value="BTC"
                                                >
                                                                 
                                                                BTC
                                                                 
                                                </option>
                                                <option
                                                                value="ETH"
                                                >
                                                                 
                                                                ETH
                                                                 
                                                </option>
                                                <option
                                                                value="XRP"
                                                >
                                                                 
                                                                XRP
                                                                 
                                                </option>
                                </select>
                                <input
                                                name="amount"
                                                onChange={[Function]}
                                                placeholder="amount"
                                                type="text"
                                                value=""
                                />
                                <input
                                                type="submit"
                                />
                </form>
</div>,
              <div
                className="right"
>
                <h2>
                                 Today's Market 
                </h2>
                <Connect(Component) />
</div>,
            ],
            "className": "body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(UserPortfolio) />,
                  <h4>
                     Add a currency you own 
</h4>,
                  <form
                    onSubmit={[Function]}
>
                    <select
                                        name="name"
                                        onChange={[Function]}
                    >
                                        <option
                                                            value="BTC"
                                        >
                                                             
                                                            BTC
                                                             
                                        </option>
                                        <option
                                                            value="ETH"
                                        >
                                                             
                                                            ETH
                                                             
                                        </option>
                                        <option
                                                            value="XRP"
                                        >
                                                             
                                                            XRP
                                                             
                                        </option>
                    </select>
                    <input
                                        name="amount"
                                        onChange={[Function]}
                                        placeholder="amount"
                                        type="text"
                                        value=""
                    />
                    <input
                                        type="submit"
                    />
</form>,
                ],
                "className": "left",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Add a currency you own ",
                  },
                  "ref": null,
                  "rendered": " Add a currency you own ",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <select
                        name="name"
                        onChange={[Function]}
>
                        <option
                                                value="BTC"
                        >
                                                 
                                                BTC
                                                 
                        </option>
                        <option
                                                value="ETH"
                        >
                                                 
                                                ETH
                                                 
                        </option>
                        <option
                                                value="XRP"
                        >
                                                 
                                                XRP
                                                 
                        </option>
</select>,
                      <input
                        name="amount"
                        onChange={[Function]}
                        placeholder="amount"
                        type="text"
                        value=""
/>,
                      <input
                        type="submit"
/>,
                    ],
                    "onSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <option
                            value="BTC"
>
                             
                            BTC
                             
</option>,
                          <option
                            value="ETH"
>
                             
                            ETH
                             
</option>,
                          <option
                            value="XRP"
>
                             
                            XRP
                             
</option>,
                        ],
                        "name": "name",
                        "onChange": [Function],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": "BTC",
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              " ",
                              "BTC",
                              " ",
                            ],
                            "value": "BTC",
                          },
                          "ref": null,
                          "rendered": Array [
                            " ",
                            "BTC",
                            " ",
                          ],
                          "type": "option",
                        },
                        Object {
                          "instance": null,
                          "key": "ETH",
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              " ",
                              "ETH",
                              " ",
                            ],
                            "value": "ETH",
                          },
                          "ref": null,
                          "rendered": Array [
                            " ",
                            "ETH",
                            " ",
                          ],
                          "type": "option",
                        },
                        Object {
                          "instance": null,
                          "key": "XRP",
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              " ",
                              "XRP",
                              " ",
                            ],
                            "value": "XRP",
                          },
                          "ref": null,
                          "rendered": Array [
                            " ",
                            "XRP",
                            " ",
                          ],
                          "type": "option",
                        },
                      ],
                      "type": "select",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "name": "amount",
                        "onChange": [Function],
                        "placeholder": "amount",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "form",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                     Today's Market 
</h2>,
                  <Connect(Component) />,
                ],
                "className": "right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Today's Market ",
                  },
                  "ref": null,
                  "rendered": " Today's Market ",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
